plugin/src/main/kotlin/com/varabyte/kobweb/intellij/notification/KobwebNotifier.kt

```diff
@@ -0,0 +1,134 @@
+            title(title)
+        }
+
+        private class ActionData(val text: String, val action: () -> Unit)
+
+        private var title: String = ""
+        private var type: NotificationType = NotificationType.INFORMATION
+        private var actionDataList = mutableListOf<ActionData>()
+
+        fun title(title: String) = apply { this.title = title }
+
+        /**
+         * Register a clickable action associated with this notification.
+         *
+         * You can register multiple actions per notification.
+         *
+         * Once clicked, the action will be performed and the notification will be dismissed.
+         */
+        fun addAction(text: String, action: () -> Unit) =
+            apply { actionDataList.add(ActionData(text, action)) }
+
+        fun type(type: NotificationType) = apply { this.type = type }
+
+        fun notify(project: Project): KobwebNotificationHandle {
+            val notification = notificationGroup.createNotification(
+                title,
+                message,
+                type,
+            )
+
+            actionDataList.forEach { actionData ->
+                notification.addAction(object : NotificationAction(actionData.text) {
+                    override fun actionPerformed(e: AnActionEvent, notification: Notification) {
+                        actionData.action()
+                        notification.expire()
+                    }
+                })
+            }
+
+            notification.notify(project)
+            return KobwebNotificationHandle(notification)
+        }
+    }
+
+    @Suppress("unused")
+    companion object {
+        fun notify(
+            project: Project,
+            title: String,
+            message: String,
+            type: NotificationType? = null
+        ): KobwebNotificationHandle {
+            return Builder(title, message)
+                .apply { type?.let { type(it) } }
+                .notify(project)
+        }
+
+        fun notify(project: Project, message: String, type: NotificationType? = null): KobwebNotificationHandle {
+            return Builder(message)
+                .apply { type?.let { type(it) } }
+                .notify(project)
+        }
+
+        fun notify(
+            project: Project,
+            title: String,
+            message: String,
+            actionText: String,
+            type: NotificationType? = null,
+            action: () -> Unit
+        ): KobwebNotificationHandle {
+            return Builder(title, message)
+                .apply { type?.let { type(it) } }
+                .addAction(actionText, action)
+                .notify(project)
+        }
+
+        fun notify(
+            project: Project,
+            message: String,
+            actionText: String,
+            type: NotificationType? = null,
+            action: () -> Unit
+        ): KobwebNotificationHandle {
+            return notify(project, title = "", message, actionText, type, action)
+        }
```

Comment:
I do not see a benefit in these helper functions over just using the Builder directly - I mean isnt that what the builder is for? It also does not enable you to write less code, and the builder pattern is more concise if one doesnt use named parameters.

Compare

```kt
KobwebNotifier.Builder(title = "", "The Kobweb plugin requires a one-time sync to enable functionality.")
                .type(NotificationType.WARNING)
                .addAction("Sync Project") {
                    // Action...
                }.notify(project)
```

to
```kt
KobwebNotifier.notify(
                project,
                "The Kobweb plugin requires a one-time sync to enable functionality.",
                "Sync Project",
                NotificationType.WARNING
            ) {
                // Action...
            }
```

Reply:
Sure, let's remove them. 

Before:

```kotlin
KobwebNotifier.notify(
    project,
    "The Kobweb plugin requires a one-time sync to enable functionality.",
    "Sync Project",
    NotificationType.WARNING
) {
    ActionUtil.invokeAction(refreshProjectAction, { dataId ->
        when {
            PlatformDataKeys.PROJECT.`is`(dataId) -> project
            else -> null
        }
    }, ActionPlaces.NOTIFICATION, null, null)
}
```

After:

```kotlin
KobwebNotifier.Builder("The Kobweb plugin requires a one-time sync to enable functionality.")
    .type(NotificationType.WARNING)
    .addAction("Sync Project") {
        ActionUtil.invokeAction(refreshProjectAction, { dataId ->
            when {
                PlatformDataKeys.PROJECT.`is`(dataId) -> project
                else -> null
            }
        }, ActionPlaces.NOTIFICATION, null, null)
    }
    .notify(project)
```

If there's *any* value in the notify helpers, it would be that you couldn't create a builder and then forget to fire it. That said, that's not worth the extra code -- you're right, simpler is better!